### PR TITLE
Fixed smoketest

### PR DESCRIPTION
**Pull request by sternj**

### CODE REVIEW

File: include/memcpysampler.hpp [hunk 1]
```diff
@@ -184,7 +189,7 @@ class MemcpySampler {
         _memcpyTriggered(0) {
           static HL::PosixLock init_lock;
           init_lock.lock();
-    sighandler_t old_sig = signal(MemcpySignal, SIG_IGN);
+    SIGHANDLER_TYPE old_sig = signal(MemcpySignal, SIG_IGN);
```
Comment (emeryberger) [inline include/memcpysampler.hpp:192]: why not just use `auto`?

File: heaplayers-make.mk [hunk 2]
```diff
@@ -2,16 +2,21 @@
 CPPFLAGS = -std=c++14 -g -O3 -DNDEBUG -fno-builtin-malloc -fvisibility=hidden
 CXX = clang++
 
+UNAME_S := $(shell uname -s)
+UNAME_P := $(shell uname -p)
+ifeq ($(UNAME_S),Darwin)
+	ifeq ($(UNAME_P),arm64)
+		ARMFLAG = -arch arm64
```
Comment (emeryberger) [inline heaplayers-make.mk:9]: great idea to do this!

File: include/sampleheap.hpp [hunk 3]
```diff
@@ -54,11 +61,11 @@ class SampleHeap : public SuperHeap {
         _freedLastMallocTrigger(false) {
       
       signal_init_lock.lock();
-      sighandler_t old_malloc = signal(MallocSignal, SIG_IGN);
+      SIGHANDLER_TYPE old_malloc = signal(MallocSignal, SIG_IGN);
```
Comment (emeryberger) [inline include/sampleheap.hpp:64]: same comment as above, just use `auto` (or `decltype` if you really need a type declaration)

File: include/sampleheap.hpp [hunk 4]
```diff
@@ -54,11 +61,11 @@ class SampleHeap : public SuperHeap {
...
       if (old_malloc != SIG_DFL) {
         signal(MallocSignal, old_malloc);
       }
-      sighandler_t old_free = signal(FreeSignal, SIG_IGN);
+      SIGHANDLER_TYPE old_free = signal(FreeSignal, SIG_IGN);
```
Comment (emeryberger) [inline include/sampleheap.hpp:68]: ditto

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: include/memcpysampler.hpp (hunk 1 above)
Comment (sternj) [reply include/memcpysampler.hpp:192]: In all honesty, I just forgot that auto existed. I'll switch it to that!

**Pull request merged**